File: 094. Binary Tree Inorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
      // Start typing your C/C++ solution below  
      // DO NOT write int main() function  
      vector<int> result;  
      inorderTra(root, result);  
      return result;  
    }
    
    void inorderTra(TreeNode* node, vector<int> &result)  
    {  
      if(node == NULL)  
      {        
        return;  
      }  
      inorderTra(node->left, result);  
      result.push_back(node->val);      
      inorderTra(node->right, result);  
    }  
};

File: 094. Binary Tree Inorder Traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def inorderTraversal(self, root):
        """
        :type root: TreeNode
        :rtype: List[int]
        """
        result = []
        stack = []
        p = root
        while p or stack:
            # Save the nodes which have left child
            while p:
                stack.append(p)
                p = p.left
            if stack:
                p = stack.pop()
                # Visit the middle node
                result.append(p.val)
                # Visit the right subtree
                p = p.right
        return result